visualization/app/codeCharta/state/store/dynamicSettings/colorRange/colorRange.reducer.ts

```diff
@@ -1,11 +1,11 @@
 import { ColorRangeAction, ColorRangeActions, setColorRange } from "./colorRange.actions"
 import { ColorRange } from "../../../../codeCharta.model"
-import _ from "lodash"
+const clone = require("rfdc")()
 
 export function colorRange(state: ColorRange = setColorRange().payload, action: ColorRangeAction): ColorRange {
 	switch (action.type) {
 		case ColorRangeActions.SET_COLOR_RANGE:
-			return _.cloneDeep(action.payload)
+			return clone(action.payload)
```

Comment:
It would be great to document for what reason the cloning is done. The code itself does not provide any hint why it's necessary. In most cases I would expect a cloning to happen right before code that mutates parts of the object. So documenting it would be good (I recommend to remove the cloning of an individual function and to run the test suite afterwards to identify all places that do not work properly anymore. Otherwise it might be difficult to identify the actual reason retrospectively).

Reply:
This is something, that I need to look into again. Setting a new state by using the whole payload should not require any cloning. However, if there still exists another reference to the colorRange object, you can mutate the state without dispatching an action into the storeService.